extensions/vertx-http/dev-ui-resources/src/main/resources/dev-ui/qwc/qwc-theme-switch.js

```diff
@@ -0,0 +1,112 @@
+import { LitElement, html, css } from 'lit';
+import { themeState } from 'theme-state';
+import { StorageController } from 'storage-controller';
+import '@vaadin/button';
+
+/**
+ * Basic theme switch
+ */
+export class QwcThemeSwitch extends LitElement {
+    storageControl = new StorageController(this);
+    
+    themes = [
+        { id: 0, name: 'Desktop', icon: 'font-awesome-solid:desktop' },
+        { id: 1, name: 'Light', icon: 'font-awesome-solid:sun' },
+        { id: 2, name: 'Dark', icon: 'font-awesome-solid:moon' }
+    ];
+    
+    static styles = css`
+        .themeButton {
+            padding-left: 10px;
+        }
+        .button {
+            --vaadin-button-background: var(--lumo-base-color);
+        }
+    `;
+
+    static properties = {
+        _selectedThemeIndex: {state: true},
+    };
+
+    constructor() {
+        super();
+        this._restoreThemePreference();
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        this._desktopTheme = "dark"; // default
+        if(window.matchMedia){
+             if(window.matchMedia('(prefers-color-scheme: light)').matches){
+                this._desktopTheme = "light";
+             }
+            
+             // Change theme setting when OS theme change
+             window.matchMedia('(prefers-color-scheme: light)').addEventListener('change', e => {
+                 if(e.matches){
+                    this._desktopTheme = "light";
+                 }else{
+                    this._desktopTheme = "dark";
+                 }
+                 if(this._selectedThemeIndex===0){
+                     this._changeToSelectedThemeIndex();
+                 }
+            });
+        }
+        
+        this._changeToSelectedThemeIndex();
+    }
+
+    render() {
+        let theme = this.themes[this._selectedThemeIndex];
+        
+        return html`<div class="themeButton">
+                        <vaadin-button theme="icon" aria-label="${theme.name}" title="${theme.name} theme" class="button" @wheel="${this._scrollTheme}" @click="${this._nextTheme}">
```

Comment:
I'm not a fan of scrolling with mousewheel here, it is too easy to do it by accident and then not know what happened or how to revert it. What do others think?

Reply:
You mouse pointer has to be on the icon, and the scroll basically go through the 3 options, so to revert you can scroll the other way, or continue to scroll.

Reply:
Yes I know that, but that's because I read the PR description. If I was a random user, I could imagine scrolling by accident and not knowing what I did :)

Reply:
(It's kind of the same as on Linux Desktop if you scroll over your Desktop switcher icon it flips between your Desktops. It really never happens on accident.

Reply:
> (It's kind of the same as on Linux Desktop if you scroll over your Desktop switcher icon it flips between your Desktops. It really never happens on accident.)

Oh it has happened to me by accident many times, and it took me a while to figure it out and start being careful about it :)

Reply:
OK lets remove it.

Reply:
Removed.